rpmbuild/copr_rpmbuild/config.py

```diff
@@ -0,0 +1,37 @@
+"""
+Configuration class for copr-rpmbuild
+"""
+
+
+import os
+import yaml
+
+
+CONFIG_PATH = "/etc/copr-rpmbuild/copr-rpmbuild"
+ALLOWED_FILE_TYPES = ["yaml", "yml"]
+
+
+class Config:
+    """
+    Configuration class for copr-rpmbuild
+    """
+    def __init__(self):
+        self.job_tag_to_mock_snippets = {}
+
+    def load_config(self):
+        """
+        Load configuration from the config file
+        """
+        config_path = None
+        for file_type in ALLOWED_FILE_TYPES:
+            if os.path.exists(f"{CONFIG_PATH}.{file_type}"):
+                config_path = f"{CONFIG_PATH}.{file_type}"
+                break
```

Comment:
There is a TOCTOU issue, and a file-not-found bug.  I don't think we need to support two suffixes, so simpler approach:

```patch
-CONFIG_PATH = "/etc/copr-rpmbuild/copr-rpmbuild"
-ALLOWED_FILE_TYPES = ["yaml", "yml"]
+CONFIG_PATH = "/etc/copr-rpmbuild/copr-rpmbuild.yml"
 
 
 class Config:
@@ -22,16 +21,10 @@
         """
         Load configuration from the config file
         """
-        config_path = None
-        for file_type in ALLOWED_FILE_TYPES:
-            if os.path.exists(f"{CONFIG_PATH}.{file_type}"):
-                config_path = f"{CONFIG_PATH}.{file_type}"
-                break
-
-        if config_path is None:
-            return
-
-        with open(CONFIG_PATH, "r", encoding="utf-8") as file:
-            config_data = yaml.safe_load(file) or {}
-
+        config_data = {}
+        try:
+            with open(CONFIG_PATH, "r", encoding="utf-8") as file:
+                config_data = yaml.safe_load(file) or {}
+        except FileNotFoundError:
+            pass
         self.job_tag_to_mock_snippets = config_data.get('job_tag_to_mock_snippets', {})
```

Reply:
> I don't think we need to support two suffixes

sooner or later someone will wonder why yaml file doesn't work because of this... I did last week :D

the bug was with CONFIG_FILE -> config_file so this should fix it... and I don't see where's the TOCTOU issue :/

Reply:
> I did last week :D

This mistake is unlikely to be repeated (we already provide the "stub" config file for in-place edists).

It's similar like with `.cfg`  vs. `.conf` vs. `.ini`.

> I don't see where's the TOCTOU issue

If file found => open it.  It may be created after doing the check (you should always open the file, and react if the file is not found).

Reply:
if file is not found in this case it'll use the default config so the end result is the same anyway

Reply:
marking as resolved (I'm fine if we agree on disagreement)